GitUI/CommandsDialogs/FormDeleteBranch.cs

```diff
@@ -62,50 +70,67 @@ private void FormDeleteBranchLoad(object sender, EventArgs e)
+            SuspendLayout();
+            MinimumSize = MaximumSize = Size.Empty;
 
-                if (_currentBranch is not null && selectedBranches.Any(branch => branch.Name == _currentBranch))
-                {
-                    MessageBox.Show(this, string.Format(_cannotDeleteCurrentBranchMessage.Text, _currentBranch), _deleteBranchCaption.Text, MessageBoxButtons.OK, MessageBoxIcon.Error);
-                    return;
-                }
+            int height = ControlsPanel.Height + MainPanel.Padding.Top + MainPanel.Padding.Bottom
+                       + tlpnlMain.Height + tlpnlMain.Margin.Top + tlpnlMain.Margin.Bottom + DpiUtil.Scale(42);
 
-                // always treat branches as unmerged if there is no current branch (HEAD is detached)
-                var hasUnmergedBranches = _currentBranch is null || selectedBranches.Any(branch => !_mergedBranches.Contains(branch.Name));
+            MinimumSize = new Size(tlpnlMain.PreferredSize.Width + DpiUtil.Scale(70), height);
+            MaximumSize = new Size(Screen.PrimaryScreen.Bounds.Width, height);
+            Size = new Size(Width, height);
+            ResumeLayout();
+        }
 
-                // we could show yes/no dialog and set forcing checkbox automatically, but more safe way is asking user to do it himself
-                if (hasUnmergedBranches && !ForceDelete.Checked)
+        private void Delete_Click(object sender, EventArgs e)
+        {
+            var selectedBranches = Branches.GetSelectedBranches().ToArray();
+            if (!selectedBranches.Any())
+            {
+                return;
+            }
+
+            if (_currentBranch is not null && selectedBranches.Any(branch => branch.Name == _currentBranch))
+            {
+                MessageBox.Show(this, string.Format(_cannotDeleteCurrentBranchMessage.Text, _currentBranch), _deleteBranchCaption.Text, MessageBoxButtons.OK, MessageBoxIcon.Error);
+                return;
+            }
+
+            if (!AppSettings.DontConfirmDeleteBranch)
+            {
+                TaskDialogPage page = new()
                 {
-                    MessageBox.Show(this, _deleteUnmergedBranchForcingSuggestion.Text, _deleteBranchCaption.Text, MessageBoxButtons.OK, MessageBoxIcon.Error);
-                    return;
-                }
+                    Text = _deleteBranchQuestion.Text,
+                    Caption = _deleteBranchConfirmTitle.Text,
+                    Icon = TaskDialogIcon.Information,
+                    Buttons = { TaskDialogButton.Yes, TaskDialogButton.No },
+                    Footnote = _useReflogHint.Text,
+                    Verification = new(TranslatedStrings.DontShowAgain),
+                    SizeToContent = true
+                };
 
-                // ask for confirmation to delete unmerged branch that may cause loosing commits
-                // (actually we could check if there are another branches pointing to that commit)
-                if (hasUnmergedBranches
```

Comment:
I would not show the confirmation dialog if all branches have been merged into HEAD.

Reply:
Done